### ISSUE TITLE
CodeMirror issues with wrapping lines

### ISSUE THREAD

**Issue by roman-r-m** [laurent22/joplin#4469]
Joplin 1.7.10 (prod, win32)

Create a note with the below text (without code block, just plain text):
```
test test

https://discourse.joplinapp.org/t/dropbox-sync-hangs-on-16-mb-large-resource/8365/7
```
(On my PC the 2n line wraps around, if not -- add some more text.)

Now set cursor before `https` and press `End` -- the cursor moves to the start on the next (wrapped line). Pressing `End` again does not do anything. (**Issue 1**)
Press `Up Arrow`, the cursor moves to the 2nd (empty) line instead of the beginning of the URL. (**Issue 2**)

Video:
![7zetbXRqwP](https://user-images.githubusercontent.com/995612/106864804-44451a00-66c2-11eb-9afc-5d8b842336c3.gif)

**Comment by CalebJohn** [comment]
This is a bug with Electron + CodeMirror. When CodeMirror is in contenteditable mode (when spellchecking is enabled), it's not possible to place the cursor at the end of a wrapped line, the browser (Electron in our case) will automatically move the cursor to the beginning of the next line. CodeMirror knows where the cursor is supposed to be (end of the line), so when you move the cursor it is correctly place at the end of the previous or next line. Similarly, if you press end again, nothing happens because you're at the end of the current line.

AFAIK there is no fix for this. But the new version of CodeMirror (CodeMirror 6) is due to be released soon, and I think this will be resolved.

TL;DR
The cursor is being drawn in the wrong place because of a bug in the way Chrome draws the cursor in contenteditable.

**Comment by lkiThakur** [comment]
> I think this will be resolved.
It's resolved I think.
in my `Joplin 1.7.11 (prod, linux)` I am unable to reproduce it.

**Comment by CalebJohn** [comment]
@lkiThakur try again with the Beta spellchecking enabled in settings.

**Comment by lkiThakur** [comment]
> @lkiThakur try again with the Beta spellchecking enabled in settings.

Yes i was able to reproduce it after enabling this option.
